jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/ExtensionStack.java

```diff
@@ -107,20 +109,37 @@ public void onFrame(Frame frame, Callback callback)
      * <p>
      * For the list of negotiated extensions, use {@link #getNegotiatedExtensions()}
      *
-     * @param configs the configurations being requested
+     * @param offeredConfigs    the configurations being requested by the client
+     * @param negotiatedConfigs the configurations accepted by the server
      */
-    public void negotiate(DecoratedObjectFactory objectFactory, ByteBufferPool bufferPool, List<ExtensionConfig> configs)
+    public void negotiate(DecoratedObjectFactory objectFactory, ByteBufferPool bufferPool, List<ExtensionConfig> offeredConfigs, List<ExtensionConfig> negotiatedConfigs)
     {
         if (LOG.isDebugEnabled())
-            LOG.debug("Extension Configs={}", configs);
+            LOG.debug("Extension Configs={}", negotiatedConfigs);
 
         this.extensions = new ArrayList<>();
 
         String rsvClaims[] = new String[3];
 
-        for (ExtensionConfig config : configs)
+        for (ExtensionConfig config : negotiatedConfigs)
         {
-            Extension ext = factory.newInstance(objectFactory, bufferPool, config);
+            Extension ext;
+
+            try
+            {
+                ext = factory.newInstance(objectFactory, bufferPool, config);
+            }
+            catch (Throwable t)
+            {
+                for (ExtensionConfig offered : offeredConfigs)
+                {
+                    if (offered.getParameterizedName().equals(config.getParameterizedName()))
+                        throw new BadMessageException("offered extension had bad parameters: ", t);
```

Comment:
@gregw this will throw this exception on both the client and server, so this probably isn't correct

perhaps i could throw some exception here and then catch it at a higher level and decide what to do

Reply:
If there is a good spot to catch and throw, then perhaps.... although BadMessage can equally apply to a response received by a client, so it's probably OK.  It's just that the exception is only meaningfully handled on the server.

Reply:
Some comments explaining the logic here would help... then I'll accept!